src/snowflake/connector/crl.py

```diff
@@ -0,0 +1,586 @@
+                # For RSA signatures, we need to use PKCS1v15 padding
+                public_key.verify(
+                    crl.signature,
+                    crl.tbs_certlist_bytes,
+                    padding.PKCS1v15(),
+                    hash_algorithm,
+                )
+            elif isinstance(public_key, ec.EllipticCurvePublicKey):
+                # For EC signatures, use ECDSA algorithm
+                public_key.verify(
+                    crl.signature,
+                    crl.tbs_certlist_bytes,
+                    ec.ECDSA(hash_algorithm),
+                )
+            else:
+                # For other key types (DSA, etc.), try without padding
+                public_key.verify(
+                    crl.signature,
+                    crl.tbs_certlist_bytes,
+                    hash_algorithm,
+                )
+
+            logger.debug("CRL signature verification successful")
+            return True
+        except Exception as e:
+            logger.warning("CRL signature verification failed: %s", e)
+            return False
+
+    def _check_certificate_against_crl(
+        self, cert: x509.Certificate, crl: x509.CertificateRevocationList
+    ) -> CRLValidationResult:
+        """Check if certificate is revoked according to CRL"""
+        revoked_cert = crl.get_revoked_certificate_by_serial_number(cert.serial_number)
+        return (
+            CRLValidationResult.REVOKED
+            if revoked_cert
+            else CRLValidationResult.UNREVOKED
+        )
+
+    def validate_connection(self, connection: SSLConnection | SSLObject) -> bool:
+        """
+        Validate an connection against CRLs.
+
+        This method extracts certificate chains from the connection and validates them
+        against Certificate Revocation Lists (CRLs).
+
+        Args:
+            connection: OpenSSL or SSLObject connection object
+
+        Returns:
+            True if validation passes, False otherwise
+        """
+        try:
+            if isinstance(connection, SSLObject):
+                # AIOHTTP SSLObject case
+                cert_chain = connection._sslobj.get_unverified_chain()
+                cert_der_chain = [
+                    ssl.PEM_cert_to_DER_cert(c.public_bytes()) for c in cert_chain
+                ]
+            else:
+                # OpenSSL SSLConnection case
+                cert_chain = connection.get_peer_cert_chain()
+                cert_der_chain = [
+                    dump_certificate(FILETYPE_ASN1, c) for c in cert_chain
+                ]
+
+            certificate_chains = self._extract_certificate_chains_from_connection(
+                cert_der_chain
+            )
+        except Exception as e:
+            logger.warning(
+                "Failed to extract certificate chain for CRL validation: %s", e
+            )
+            certificate_chains = []
+        return self.validate_certificate_chains(certificate_chains)
```

Comment:
This method has changed compared to main branch. The if statement handles sync vs aio scenario. We can potentially refactor it by accepting only `certificate_chains` and delegating certificate pre-processing to other places.